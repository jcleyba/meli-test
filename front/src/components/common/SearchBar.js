import React, { Component } from 'react';

import './SearchBar.css';

class SearchBar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      term: this.props.term || ''
    };
  }

  onInputChange = (event) => {
    this.setState({
      term: event.target.value
    })
  };

  onSubmit = (event) => {
    event.preventDefault();
    this.props.onSubmit(this.state.term)
  };

  render() {
    return (
      <div className="search-bar">
        <form onSubmit={this.onSubmit}>
          <input name='search' type="text" className="search-input" placeholder="Nunca dejes de buscar"
                 onChange={this.onInputChange}
                 value={this.state.term}/>
          <button type="submit" className="search-btn"></button>
        </form>
      </div>
    );
  }
}

export default SearchBar;
