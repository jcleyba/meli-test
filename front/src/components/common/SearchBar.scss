@import '../../styles/_variables';

.search-bar {
  width: 500px;
  display: flex;
  align-items: center;
  form,
  .search-input {
    width: 100%;
    display: flex;
  }
  .search-input {
    font-size: 18px;
    color: $grey;
    padding: 5px 60px 5px 15px;
    border: none;
    border-radius: 3px 0 0 3px;
  }
  .search-btn {
    cursor: pointer;
    border-radius: 0 3px 3px 0;
    width: 46px;
    height: 40px;
    border: none;
    background: url("../../images/ic_Search@2x.png.png") no-repeat center $light_grey;
    background-size: 20px;
  }
}
