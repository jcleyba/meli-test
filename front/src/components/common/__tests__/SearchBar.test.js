import React from 'react';
import { shallow, mount } from 'enzyme';

import 'jest-enzyme';

import SearchBar from '../SearchBar';

describe('SearchBar Component', () => {

  let wrapper;
  let mounted;

  const onFormSubmitSpy = jest.fn();

  const props = {
    onSubmit: jest.fn()
  };


  beforeEach(() => {
    wrapper = shallow(<SearchBar {...props}/>);
    mounted = mount(<SearchBar {...props}/>);

    mounted.instance().onFormSubmit = onFormSubmitSpy;
    mounted.update();
  });

  afterEach(() => {
    wrapper = null;
    mounted = null;

    onFormSubmitSpy.mockReset();
  });

  describe('SearchBar', () => {

    it('matches current snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have onSubmit prop', () => {
      expect(mounted).toHaveProp('onSubmit');
    });

    it('should a term in state', () => {
      expect(mounted).toHaveState('term');
    });

    it('should have className of search-bar', () => {
      expect(wrapper).toHaveClassName('search-bar');
    });

    it('should have an input', () => {
      expect(mounted.find('input')).toBePresent();
    });

    it('should have a button', () => {
      expect(mounted.find('button')).toBePresent();
    });

    it('call function when submitted', () => {
      const preventDefault = jest.fn();
      wrapper.find('form').simulate('submit', {preventDefault});

      expect(preventDefault).toHaveBeenCalled();
    });

    it('pass value onChange to input', () => {
      wrapper.find('.search-input').simulate('change', {
        target: {
          value: 'ipad'
        }
      });

      expect(wrapper).toMatchSnapshot();
    });
  })

});
