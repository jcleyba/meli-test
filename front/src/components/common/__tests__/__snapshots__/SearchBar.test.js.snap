// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component SearchBar matches current snapshot 1`] = `"<div class=\\"search-bar\\"><form><input type=\\"text\\" name=\\"search\\" class=\\"search-input\\" placeholder=\\"Nunca dejes de buscar\\" value=\\"\\"/><button type=\\"submit\\" class=\\"search-btn\\"></button></form></div>"`;

exports[`SearchBar Component SearchBar pass value onChange to input 1`] = `
<div
  className="search-bar"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="search-input"
      name="search"
      onChange={[Function]}
      placeholder="Nunca dejes de buscar"
      type="text"
      value="ipad"
    />
    <button
      className="search-btn"
      type="submit"
    />
  </form>
</div>
`;
